### PR TITLE
Handle 'Fatal: 400 Bad Request' while removing backup resource from virtual server and updating job on veean side.

### PR DESCRIPTION
**Pull request by ykohut**
This is to solve CORE-16412.

### CODE REVIEW

File: remove_virtual_server.rb [hunk 1]
```diff
@@ -1,7 +1,13 @@
 Backups::Plugin.hook helpers: %i[client_helper] do
   def call(virtual_server)
     virtual_server.metadata[:veeam_related_job_ids]&.each do |job_id|
-      api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
+      begin
```
Comment (hasrthur) [inline remove_virtual_server.rb:4]:
```suggestion
        beging
          api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
        rescue => err
          logger.error(err.message)
        end
```

File: remove_virtual_server.rb [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+          api_get("jobs/#{job_id}")
+      rescue => err
+          # Something went wrong while matching Veeam Job for the virtual server
+      else
+          api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
+      end
```
Comment (lesykgg) [inline remove_virtual_server.rb]:
```ruby
begin
  api_get("jobs/#{job_id}")

  api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
rescue RestClient::BadRequest
  # do nothing
end
```

Comment (ykohut) [reply remove_virtual_server.rb]: Done.

**Review by lesykgg — APPROVED** [review]

File: remove_virtual_server.rb (hunk 2 above)
Comment (nahaylo) [reply remove_virtual_server.rb]: But in case when `api_put` fails, you will rescue that request and won't do nothing. Is that expected behavior?

**Review by nahaylo — APPROVED** [review]

File: remove_virtual_server.rb (hunk 2 above)
Comment (ykohut) [reply remove_virtual_server.rb]: Yep, if request fails, we just do nothing at Veeam side, and disable the resource for VM in OnApp.

File: remove_virtual_server.rb [hunk 3]
```diff
@@ -1,7 +1,11 @@
 Backups::Plugin.hook helpers: %i[client_helper] do
   def call(virtual_server)
     virtual_server.metadata[:veeam_related_job_ids]&.each do |job_id|
-      api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
+      begin
+          api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
+      rescue RestClient::BadRequest
+          # do nothing
```
Comment (hasrthur) [inline remove_virtual_server.rb]:
```suggestion
          logger.error("Job #{job_id} doesn't exist on third-party")
```

**Review by hasrthur — APPROVED** [review]

File: remove_virtual_server.rb (hunk 3 above)
Comment (ykohut) [reply remove_virtual_server.rb]: Done.
Please provide me with write access to the repository to merge.
Also, please enable "Allow squash merging" for the repository.

Comment (jwithall) [reply remove_virtual_server.rb]: Allow squash merging - enabled
Write access granted for ykohut - let me know if any problems

File: remove_virtual_server.rb [hunk 4]
```diff
@@ -1,7 +1,11 @@
 Backups::Plugin.hook helpers: %i[client_helper] do
   def call(virtual_server)
     virtual_server.metadata[:veeam_related_job_ids]&.each do |job_id|
-      api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
+      begin
+          api_put("jobs/#{job_id}", disable_schedule_params.to_xml)
+      rescue RestClient::BadRequest => err
+          logger.error("Job #{job_id} doesn't exist on third-party: " + err.message)
```
Comment (hasrthur) [inline remove_virtual_server.rb]: logger.error("Job #{job_id} doesn't exist on third-party: #{err.message}")

**Review by hasrthur — APPROVED** [review]

**Pull request merged**